frontend/src/components/GoalCards/GoalCard.js

```diff
@@ -254,23 +254,127 @@ export default function GoalCard({
   };
 
   const renderEnteredBy = () => {
-    let specialists;
+    const tags = [];
+
     if (isMonitoringGoal) {
-      specialists = [
-        {
-          name: 'System-generated',
-          roles: ['OHS'],
-        },
-      ];
-    } else if (collaborators) {
-      specialists = collaborators
-        .filter((c) => c.goalCreatorName)
-        .map((c) => ({
-          name: c.goalCreatorName,
-          roles: [c.goalCreatorRoles].flat(),
-        }));
+      // monitoring goals: show a single "System-generated" tag with "OHS" role
+      tags.push(
+        <Tag key="system-generated-tag" clickable>
+          <Tooltip
+            displayText="OHS"
+            screenReadDisplayText={false}
+            buttonLabel="reveal the full name of this user"
+            tooltipText="System-generated"
+            hideUnderline
+            buttonClassName="display-flex"
+            className="ttahub-goal-card__entered-by-tooltip"
+          />
+        </Tag>,
+      );
+    } else if (collaborators && collaborators.length > 0) {
+      const creatorsWithName = collaborators.filter((c) => c.goalCreatorName);
+
+      if (creatorsWithName.length > 0) {
+        creatorsWithName.forEach((c) => {
+          if (c.goalCreatorRoles) {
+            // convert roles to array, handling different possible formats
+            let rolesArray = [];
+
+            if (typeof c.goalCreatorRoles === 'string') {
+              // e.g. "GS, ECS"
+              rolesArray = c.goalCreatorRoles.split(',').map((r) => r.trim());
+            } else if (Array.isArray(c.goalCreatorRoles)) {
+              // already an array
+              rolesArray = c.goalCreatorRoles;
+            } else {
+              // something else, try to flatten it
+              rolesArray = [c.goalCreatorRoles].flat().filter(Boolean);
+            }
+
+            // separate tag for each role
+            if (rolesArray.length > 0) {
+              rolesArray.forEach((role) => {
+                tags.push(
+                  <Tag key={`${c.goalCreatorName}-${role}`} clickable>
```

Comment:
Could we add the SpecialistTag here with a single array entry and any other custom props we need?

Either way I guess its fine still will need coverage for each IF path.

Reply:
see [31b9835](https://github.com/HHS/Head-Start-TTADP/pull/2781/commits/31b98350e1c4a4bcd6f9ec85390e45697a59ce43) - I moved this logic into SpecialistTags as suggested